101-cassandra-existing-vnet/azuredeploy.json

```diff
@@ -0,0 +1,281 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmSku": {
+      "type": "string",
+      "defaultValue": "Standard_DS3_v2",
+      "metadata": {
+        "description": "Size of Cassandra VMs."
+      }
+    },
+    "clusterName": {
+      "type": "string",
+      "metadata": {
+        "description": "Globally unique dns name for the Cassandra nodes scale set. Must be 3-61 characters in length and globally unique across Azure."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Admin username on all VMs."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password on all VMs."
+      }
+    },
+    "nodeCount": {
+      "type": "int",
+      "metadata": {
+        "description": "Number of VM instances (100 or less)."
+      },
+      "defaultValue": 2
+    },
+    "cassandraVersion": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      },
+      "defaultValue": "latest",
+      "allowedValues": [
+        "2.1.18",
+        "2.1",
+        "2.2.10",
+        "2.2",
+        "3.0.14",
+        "3.0",
+        "3.11.0",
+        "3.11",
+        "3",
+        "latest"
+      ]
+    },
+    "existingVirtualNetworkName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the existing VNET"
+      }
+    },
+    "existingVirtualNetworkResourceGroup": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the existing VNET resource group"
+      }
+    },
+    "existingSubnetName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the subnet in the virtual network you want to use"
+      }
+    }
+  },
+  "variables": {
+    "publicIPAddressName": "loadBalancerPublicIP",
+    "loadBalancerName": "cassandraLoadBalancer",
+    "loadBalancerFrontEndName": "loadBalancerFrontEnd",
+    "loadBalancerBackEndName": "loadBalancerBackEnd",
+    "loadBalancerProbeName": "loadBalancerHttpProbe",
+    "loadBalancerNatPoolName": "loadBalancerNatPool",
+    "namingInfix": "[toLower(parameters('clusterName'))]",
+    "nicName": "[concat(variables('namingInfix'), '-nic')]",
+    "_artifactsLocation": "https://raw.githubusercontent.com/yaron2/azure-quickstart-templates/cassandra-cluster/101-cassandra-existing-vnet/scripts",
+    "_extensionScript": "cassandraSetup.sh",
+    "nestedTemplateLocation": "https://raw.githubusercontent.com/yaron2/azure-quickstart-templates/cassandra-cluster/101-cassandra-existing-vnet/nested/cassandra-extension-vmss.json",
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts